pkg/gofr/service/health.go

```diff
@@ -16,14 +18,20 @@ type Health struct {
 }
 
 func (h *httpService) HealthCheck(ctx context.Context) *Health {
-	return h.getHealthResponseForEndpoint(ctx, ".well-known/alive")
+	return h.getHealthResponseForEndpoint(ctx, ".well-known/alive", defaultTimeout)
 }
 
-func (h *httpService) getHealthResponseForEndpoint(ctx context.Context, endpoint string) *Health {
+func (h *httpService) getHealthResponseForEndpoint(_ context.Context, endpoint string, timeout int) *Health {
 	var healthResponse = Health{
 		Details: make(map[string]interface{}),
 	}
 
+	// caeate a new context with 5 second timeout for healthCheck call.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout)*time.Second)
```

Comment:
That's the old way (before Go 1.21) to do it. I would discourage doing this.

Please consider the following code


```suggestion
	ctx, cancel := context.WithTimeout(context.WithoutCancel(ctx), time.Duration(timeout)*time.Second)
```

https://pkg.go.dev/context#WithoutCancel

Thus way all data set in the context with WithValue are still in the context, you just create a context that won't be canceled. Then you set up a timeout

Reply:
Thanks for pointing this out! We were using new context for health check calls since we did not want to have the whole health check call getting canceled by just one timing out downstream service.